configs/components/ruby-3.2.4.rb

```diff
@@ -50,8 +50,6 @@
     pkg.apply_patch "#{base}/windows_mingw32_mkmf.patch"
     pkg.apply_patch "#{base}/windows_nocodepage_utf8_fallback_r2.5.patch"
     pkg.apply_patch "#{base}/ruby-faster-load_32.patch"
-    pkg.apply_patch "#{base}/revert_speed_up_rebuilding_loaded_feature_index.patch"
-    pkg.apply_patch "#{base}/revert-ruby-double-load-symlink.patch"
```

Comment:
Can you expand on why these patches were removed?

Reply:
The reason I ask is because there is a significant perf issue on Windows and verifying it's fixed is fairly involved.

Reply:
Hey @joshcooper,
Sorry, my bad. I completed misunderstood the patches and concluded that they are not needed, so deleted them. 
The patches were failing to apply ruby 3.2.4 as the line numbers are different due to changes made to the patch files since then (we use fuzz=0 which makes the line number tolerance stricter I think).
I have reinstated the patches and made an update to fix that issue. Currently running the windows builds for agent-runtime-main and pdk-runtime to verify.
Please take a look now and let me know what you think.

Marking this as draft for now until the test runs complete.

Reply:
No worries @shubhamshinde360 thanks for running all the builds through. The changes look good to me (as long as the patches apply cleanly).

Not sure if this something we want to do in the future ruby bumps, but it's possible to pass `fuzz` in the `apply_patch` DSL method like https://github.com/puppetlabs/vanagon/blob/e344e9bb57300eb1ad8f92f3c75bb540f3e1df71/spec/lib/vanagon/component/dsl_spec.rb#L245